src/Aspire.Hosting.Azure.EventHubs/AzureEventHubsExtensions.cs

```diff
@@ -118,15 +148,13 @@ public static IResourceBuilder<AzureEventHubsResource> RunAsEmulator(this IResou
             return builder;
         }
 
-        // Add emulator container
-        var configHostFile = Path.GetTempFileName();
-        if (!OperatingSystem.IsWindows())
-        {
-            File.SetUnixFileMode(configHostFile,
-                UnixFileMode.UserRead | UnixFileMode.UserWrite
-                | UnixFileMode.GroupRead | UnixFileMode.GroupWrite
-                | UnixFileMode.OtherRead | UnixFileMode.OtherWrite);
-        }
+        var configHostFile = Path.Combine(Directory.CreateTempSubdirectory("AspireEventHubsEmulator").FullName, "Config.json");
+
+        var customMountAnnotation = new ContainerMountAnnotation(
+                configHostFile,
+                AzureEventHubsEmulatorResource.EmulatorConfigJsonPath,
+                ContainerMountType.BindMount,
+                isReadOnly: false);
```

Comment:
Why does this need to be writable?

Reply:
Custom `JsonNode` callbacks can update it.

Reply:
They don't update via the bind mount though. They update the file on the host's disk.

Reply:
Fixed